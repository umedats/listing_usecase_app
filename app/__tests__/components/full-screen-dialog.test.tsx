import { render, screen } from "@testing-library/react";
import FullScreenDialog from "@/app/components/full-screen-dialog";

describe("Test for the FullScreenDialog", () => {
  test("verifies the dialog renders with open state set to true", () => {
    render(
      <FullScreenDialog
        isOpen={true}
        onClose={() => {}}
        title={"Title"}
        closeButtonHelpText={"Close"}
      >
        <h1> Body</h1>
      </FullScreenDialog>,
    );

    expect(screen.queryByTestId("fullScreenDialog")).toMatchSnapshot();
  });

  test("verifies the dialog renders with open state set to false", () => {
    render(
      <FullScreenDialog
        isOpen={false}
        onClose={() => {}}
        title={"Title"}
        closeButtonHelpText={"Close"}
      >
        <h1> Body</h1>
      </FullScreenDialog>,
    );

    expect(screen.queryByTestId("fullScreenDialog")).toBeNull();
  });
});
