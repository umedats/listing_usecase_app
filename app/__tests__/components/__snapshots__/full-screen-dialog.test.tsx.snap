// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the FullScreenDialog verifies the dialog renders with open state set to true 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="fullScreenDialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r0:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen css-m9glnp-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-10ghrmp-MuiPaper-root-MuiAppBar-root"
        style="padding-right: 1024px;"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
        >
          <button
            aria-label="Close"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            data-testid="cancelFullScreenDialog"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiTypography-root MuiTypography-h6 css-18vlf2e-MuiTypography-root"
          >
            Title
          </div>
        </div>
      </header>
      <div
        class="MuiDialogContent-root MuiDialogContent-dividers css-1t4vnk2-MuiDialogContent-root"
      >
        <h1>
           Body
        </h1>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
